core/src/main/java/discord4j/core/object/entity/channel/AllowedMentions.java

```diff
@@ -0,0 +1,222 @@
+package discord4j.core.object.entity.channel;
+
+import discord4j.discordjson.json.AllowedMentionsData;
+import discord4j.discordjson.json.ImmutableAllowedMentionsData;
+import discord4j.discordjson.possible.Possible;
+import discord4j.rest.util.Snowflake;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * A class for holding the allowed_mentions object with an built-in factory for default values.
+ * Also this class wraps the {@link AllowedMentionsData} JSON to a Discord4J class.
+ */
+public class AllowedMentions {
+
+    private static Possible<List<AllowedMentions.Type>> DEFAULT_TYPES = Possible.absent();
+    private static Possible<List<Snowflake>> DEFAULT_USERS = Possible.absent();
+    private static Possible<List<Snowflake>> DEFAULT_ROLES = Possible.absent();
+
+    /**
+     * Crates a builder for this {@link AllowedMentions} class
+     * @return A builder class for allowed mentions
+     */
+    public static AllowedMentions.Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Sets the default types which are parsed in allowed mentions
+     * @param types the types to allow by default
+     */
+    public static void setDefaultParsedTypes(Possible<List<AllowedMentions.Type>> types) {
+        DEFAULT_TYPES = types;
+    }
+
+    /**
+     * Sets the default types which are parsed in allowed mentions
+     * @param types the types to allow by default
+     */
+    public static void setDefaultParsedTypes(List<AllowedMentions.Type> types) {
+        setDefaultParsedTypes(Possible.of(types));
+    }
+
+    /**
+     * Sets the default users which are allowed in mentions
+     * @param users the users to allow by default
+     */
+    public static void setDefaultAllowedUsers(Possible<List<Snowflake>> users) {
+        DEFAULT_USERS = users;
+    }
+
+    /**
+     * Sets the default users which are allowed in mentions
+     * @param users the users to allow by default
+     */
+    public static void setDefaultAllowedUsers(List<Snowflake> users) {
+        setDefaultAllowedUsers(Possible.of(users));
+    }
+
+    /**
+     * Sets the default roles which are allowed in mentions
+     * @param roles the roles to allow by default
+     */
+    public static void setDefaultAllowedRoles(Possible<List<Snowflake>> roles) {
+        DEFAULT_ROLES = roles;
+    }
+
+    /**
+     * Sets the default roles which are allowed in mentions
+     * @param roles the roles to allow by default
+     */
+    public static void setDefaultAllowedRoles(List<Snowflake> roles) {
+        setDefaultAllowedRoles(Possible.of(roles));
+    }
+
+    private final Possible<List<AllowedMentions.Type>> parse;
+    private final Possible<List<Snowflake>> userIds;
+    private final Possible<List<Snowflake>> roleIds;
```

Comment:
Should these be Sets?

Reply:
Yeah this could be sets, thats true. I'm going to fix that.